ui/src/criteria/attribute.tsx

```diff
@@ -0,0 +1,153 @@
+  id: number;
+  name: string;
+};
+
+interface Config extends CriteriaConfig {
+  attribute: string;
+}
+
+interface Data extends Config {
+  selected: Selection[];
+}
+
+type AttributeEditProps = {
+  dispatchFn: (data: Data) => void;
+  data: Data;
+};
+
+@registerCriteriaPlugin("attribute", (config: CriteriaConfig) => ({
+  ...(config.plugin as Config),
+  selected: [],
+}))
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+class _ implements CriteriaPlugin<Data> {
+  public data: Data;
+
+  constructor(public id: string, data: unknown) {
+    this.data = data as Data;
+  }
+
+  renderEdit(dispatchFn: (data: Data) => void) {
+    return <AttributeEdit dispatchFn={dispatchFn} data={this.data} />;
+  }
+
+  renderDetails() {
+    return <ConceptDetails data={this.data} />;
+  }
+
+  generateFilter() {
+    if (this.data.selected.length === 0) {
+      return null;
+    }
+
+    const operands = () =>
+      this.data.selected.map(({ id }) => ({
+        binaryFilter: {
+          attributeVariable: {
+            variable: "person",
+            name: this.data.attribute,
+          },
+          operator: tanagra.BinaryFilterOperator.Equals,
+          attributeValue: {
+            int64Val: id,
+          },
+        },
+      }));
+
+    return {
+      arrayFilter: {
+        operands: operands(),
+        operator: tanagra.ArrayFilterOperator.Or,
+      },
+    };
+  }
+
+  occurrenceEntities() {
+    return [];
+  }
+}
+
+function AttributeEdit(props: AttributeEditProps) {
+  const underlay = useUnderlay();
+  const attributeName = props.data.attribute;
+
+  const hintDisplayName = (hint: tanagra.EnumHintValue) =>
+    hint.displayName || "Unknown Value";
+
+  const enumHintValues = underlay.entities
+    .find((g) => g.name === underlay.primaryEntity)
+    ?.attributes?.find((attribute) => attribute.name === attributeName)
+    ?.attributeFilterHint?.enumHint?.enumHintValues;
+
+  const selectionIndex = (hint: tanagra.EnumHintValue) =>
+    props.data.selected.findIndex(
+      (row) => row.id === hint.attributeValue?.int64Val
+    );
+
+  if (enumHintValues?.length === 0) {
+    return <Typography>No attribute information!</Typography>;
+  } else {
+    return (
+      <>
+        {enumHintValues?.map((hint: tanagra.EnumHintValue) => (
+          <ListItem key={hintDisplayName(hint)}>
+            <FormControlLabel
+              label={hintDisplayName(hint)}
+              control={
+                <Checkbox
+                  size="small"
+                  checked={selectionIndex(hint) > -1}
+                  onChange={() => {
+                    props.dispatchFn(
+                      produce(props.data, (data) => {
+                        if (selectionIndex(hint) > -1) {
+                          data.selected.splice(selectionIndex(hint), 1);
+                        } else {
+                          data.selected.push({
+                            id: hint.attributeValue?.int64Val as number,
```

Comment:
`as number` was required in concept.tsx because the table data could be multiple types. Here you don't need it because `int64Val` is always a number.

Reply:
Good to learn us `|| -1` replace `as number`. 

Note: the `|| -1` can cover the case which is undefined 
but `as number` will hiding the undefined case will cause bug in the future and hard to figure out.